### DISCUSSION TITLE
Possible to upgrade Teensy to 4.x??

### DISCUSSION THREAD

**Discussion by noodleboy91** [eh2k/squares-and-circles#37] [Q&A]
Is it possible to swap out Teensy 3.2 on a 1U O_c (or a 3U 8HP version for that matter) for a 4.x one? Also what would be involved in the process (simple pull and swap, soldering, adding headers/breakouts etc?

**Comment by eh2k** [comment]
You can just replace the Teensy 3.2 with a Teensy 4.0.

If the POGO Pin is soldered - cover the bottom of the teensy with insulating tape - all other pins are compatible (see pjrc). Be careful with connecting USB and power at the same time - if you have VIN/VUSB connected.

POGO Pin: https://www.modwiggler.com/forum/viewtopic.php?p=2867702#p2867702

**Reply by noodleboy91** [reply]
Sweet, I went ahead and got one, put some headers on my o_C (cleared the pogo pin by doing this also) so I could swap between the Teensy 3.2 and 4.0 until I get another one, and loaded SnC on it, this thing is awesome!! Thank you so much... follow up question, is it possible to mix all the outputs to one output jack and/or send clock out on one of the output jacks?

**Reply by eh2k** [reply]
Hi, thanks for the feedback.

At the moment it is not possible to assign an engine to arbitary output. See https://github.com/eh2k/squares-and-circles/issues/6.

But I am still developing the firmware for my module based on the Daisy submodule https://github.com/eh2k/squares-and-circles/discussions/24 . There the routing will be different, as I only have one stereo audio output. On the O_C it would be also conceivable where audio engines are mixed to one (stereo) output, and CV engines are output separately.

**Reply by noodleboy91** [reply]
Okay no worries, I'll just use a mixer :) Definitely going to have to check out the Daisy based module as well. Also, wondering if you will be adding any of the engines from the [Brains Reloaded](https://www.synthtopia.com/content/2023/03/02/behringer-brains-reloaded-adds-5-new-modes/) update (specifically bassline lol)?

**Comment by eh2k** [comment]
@noodleboy91 

good hint - i did some research - it looks like all additions from behringer are closed source. Unfortunately mutable has also not published the last plates update.

Regarding DX7 engine, there is some information or even source code available in the internet - if I find the time, I think a DX7 engine would be quite feasible.

Regarding "basseline" - I have experimented with "open303" by Robin Schmidt (https://github.com/RobinSchmidt/Open303) some time ago. I was a bit dissatisfied with the sound - maybe because I compared it directly with my xoxbox. If there is any interest in coding, I could share my adapted source code.

**Reply by noodleboy91** [reply]
Thank you for taking them into consideration/ It is crappy that it is closed source, especially when the rest if the firmware is not :( 

The DX7 would be cool to have :)

Reguardimg the 303, the version in Brains doesnt hold a candle to even my TD3, but still cool that it is there :)

I've not coded much in years (last program I coded was in Visual Basic lol), but might be interested in giving it a go, if for nothng but to see if I cann actually do it lol.
